### PR TITLE
feat: EXPOSED-296 Add ability to check if a Sequence exists in a database

### PR DESCRIPTION
**Pull request by joc-a**
Added two functions:
- `sequences()` returns all the existing sequences that a user has the right to access in a database
- `exists()` for a `Sequence` that checks if a sequence exists by looking up all the existing sequences that a user has the right to access in a database, and checking if the identifier of the `Sequence` matches that of any of the existing sequences

### CODE REVIEW

File: exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/SequencesTests.kt [hunk 1]
```diff
@@ -132,6 +133,21 @@ class SequencesTests : DatabaseTestsBase() {
...
+                try {
+                    SchemaUtils.createSequence(myseq)
+
+                    assertTrue(myseq.exists())
+                } finally {
+                    SchemaUtils.dropSequence(myseq)
+                }
+            }
+        }
+    }
```
Comment (bog-walk) [inline exposed-tests/src/test/kotlin/org/jetbrains/exposed/sql/tests/shared/ddl/SequencesTests.kt:149]: `existingSequences()` will also be accessible from the public API via the current dialect, so it'd be good to test too.

It'd be great to test that both new functions cover manually created sequences (as done above), as well as autogenerated sequences. Both when an explicit sequence name is provided to `autoIncrement()` and when one isn't (for Oracle, that the existing result matches the result of `autoincSeq`).

File: exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt [hunk 2]
```diff
@@ -285,6 +285,43 @@ class JdbcDatabaseMetadataImpl(database: String, val metadata: DatabaseMetaData)
...
+                        sequences.add(rs.getString("SEQUENCE_NAME"))
+                    }
+                }
+            }
+            else -> {
+                val rs = metadata.getTables(null, null, null, arrayOf("SEQUENCE"))
+                while (rs.next()) {
+                    sequences.add(rs.getString(3))
+                }
+            }
```
Comment (bog-walk) [inline exposed-jdbc/src/main/kotlin/org/jetbrains/exposed/sql/statements/jdbc/JdbcDatabaseMetadataImpl.kt:319]: For the dialects that don't rely on jdbc metadata, should maybe `VendorDialect.existingSequences()` be overriden in each dialect-specific file and the sql logic extracted?
That way only the default vendor function invokes this module's function, so it continues to only contain metadata logic?

**Review by bog-walk — CHANGES_REQUESTED** [review]
Please add KDocs for the new functions. Also left a few comments.

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/api/ExposedDatabaseMetadata.kt [hunk 3]
```diff
@@ -67,6 +67,9 @@ abstract class ExposedDatabaseMetadata(val database: String) {
     /** Returns a map with the [PrimaryKeyMetadata] in each of the specified [tables]. */
     abstract fun existingPrimaryKeys(vararg tables: Table): Map<Table, PrimaryKeyMetadata?>
 
+    /** Returns a list of the names of all sequences in the database. */
+    abstract fun existingSequences(): List<String>
```
Comment (e5l) [inline exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/api/ExposedDatabaseMetadata.kt:71]: It looks like `existing` is an obsolete word in these kinds of functions

**Review by e5l — APPROVED** [review]
lgtm

**Review by bog-walk — APPROVED** [review]

File: exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/api/ExposedDatabaseMetadata.kt (hunk 3 above)
Comment (joc-a) [reply exposed-core/src/main/kotlin/org/jetbrains/exposed/sql/statements/api/ExposedDatabaseMetadata.kt:71]: Not sure I got what you mean. Do you mean in Exposed? We're using it in several other functions. Could you please clarify?

**Pull request merged**